client/components/data/query-post-stats/index.jsx

```diff
@@ -1,80 +1,42 @@
-import { isEqual } from 'lodash';
 import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
 import { requestPostStats } from 'calypso/state/stats/posts/actions';
 import { isRequestingPostStats } from 'calypso/state/stats/posts/selectors';
 
-class QueryPostStats extends Component {
-	static defaultProps = {
-		requestPostStats: () => {},
-		heartbeat: 0,
-	};
+function useMemoizedFields( fields ) {
+	const memoizedFields = useRef();
 
-	static propTypes = {
-		siteId: PropTypes.number,
-		postId: PropTypes.number,
-		fields: PropTypes.array,
-		requestingPostStats: PropTypes.bool,
-		requestPostStats: PropTypes.func,
-		heartbeat: PropTypes.number,
-	};
-
-	// @TODO: Please update https://github.com/Automattic/wp-calypso/issues/58453 if you are refactoring away from UNSAFE_* lifecycle methods!
-	UNSAFE_componentWillMount() {
-		const { requestingPostStats, siteId, postId } = this.props;
-		if ( ! requestingPostStats && siteId && typeof postId !== 'undefined' ) {
-			this.requestPostStats( this.props );
-		}
-	}
-
-	componentWillUnmount() {
-		this.clearInterval();
+	if ( fields.join() !== memoizedFields.current?.join() ) {
```

Comment:
One more thing: `<QueryPostStats />` is often rendered with no `fields` prop at all so you need to account for an `undefined` value. Means calling `fields?.join()` here.

Reply:
Great catch, fixed in https://github.com/Automattic/wp-calypso/pull/60026/commits/26bae36b1ad7897f86de0ff2b1497ab0707b42b7 👍🏻